File: Data Structure/String/Easy/LET_no1629_Slowest Key.py
class Solution:
    def slowestKey(self, releaseTimes: List[int], keysPressed: str) -> str:
        time, res = 0, ''
        releaseTimes = [0] + releaseTimes
        
        for i in range(1, len(releaseTimes)):
            diff = releaseTimes[i] - releaseTimes[i - 1]
            c = keysPressed[i - 1]
            if diff > time or diff == time and ord(c) - ord('a') > ord(res) - ord('a'):
                time = diff
                res = c
        
        return res

File: Data Structure/String/Easy/LET_no1629_Slowest Key.cpp
class Solution {
public:
    char slowestKey(vector<int>& releaseTimes, string keysPressed) {
        vector<int> times = {0};
        times.insert(times.end(), releaseTimes.begin(), releaseTimes.end());
        
        int time = 0;
        char res;
        for (int i = 1; i < times.size(); i++) {
            if (times[i] - times[i - 1] > time || times[i] - times[i - 1] == time && keysPressed[i - 1] > res) {
                time = times[i] - times[i - 1];
                res = keysPressed[i - 1];
            }
        }
        
        return res;
    }
};